content/docs/cucumber/configuration.md

```diff
@@ -15,36 +15,136 @@ polyglot:
 # Type Registry
 
 {{% block "java,kotlin" %}}
-Parameter types let you convert parameters from cucumber-expressions to objects. Data table and doc string string
-types let you convert data tables and doc strings to objects. Like step definitions type definitions are part of the
-glue. When placed on the glue path Cucumber will detect them automatically.
-
-For instance, the following class registers a `ParameterType` of type Color, and a `DataTableType` of type Person:
+Parameter types let you convert parameters from cucumber-expressions to objects. Data table and doc string string types let you convert data tables and doc strings to objects. Like step definitions, type definitions are part of the glue. When placed on the glue path Cucumber will detect them automatically.
+For example, the following class registers custom "Author" data table type, "List<Book>" docstring type and "Book" parameter type:
```

Comment:
This example tries to show quite a few things and I feel that without the matching Gherkin it will be hard to understand. Let's split this up in 3 distinct examples.